src/conn/conn_compact.c

```diff
@@ -114,29 +117,35 @@ __compact_server(void *arg)
             continue;
         }
 
+        /* Make a copy of the key has it can be freed once the cursor is released. */
+        __wt_free(session, uri);
+        WT_ERR(__wt_strndup(session, key, strlen(key), &uri));
+
         /* Always close the metadata cursor. */
         WT_ERR(__wt_metadata_cursor_release(session, &cursor));
 
         /* Compact the file with the latest configuration. */
-        __wt_free(session, config);
         __wt_spin_lock(session, &conn->background_compact.lock);
-        ret = __wt_strndup(session, conn->background_compact.config,
-          conn->background_compact.config == NULL ? 0 : strlen(conn->background_compact.config),
-          &config);
+        if (config == NULL || !WT_STREQ(config, conn->background_compact.config)) {
+            __wt_free(session, config);
+            ret = __wt_strndup(session, conn->background_compact.config,
+              strlen(conn->background_compact.config), &config);
+        }
         __wt_spin_unlock(session, &conn->background_compact.lock);
 
         WT_ERR(ret);
 
-        ret = wt_session->compact(wt_session, key, config);
+        ret = wt_session->compact(wt_session, uri, config);
 
         /* FIXME-WT-11343: compaction is done, update the data structure for this table. */
         /*
          * Compact may return:
-         * - EBUSY and WT_ROLLBACK for various reasons.
+         * - EBUSY or WT_ROLLBACK for various reasons.
          * - ETIMEDOUT if the timer has been configured and compaction took too long.
-         * - WT_NOTFOUND if the underlying file has been deleted.
+         * - ENOENT or WT_NOTFOUND if the underlying file does not exist.
```

Comment:
What are the different cases for compact returning ENONENT as opposed to WT_NOTFOUND?

Reply:
I am actually not convinced we can get `WT_NOTFOUND` but rather `ENONENT` instead when the file does not exist.

Reply:
It seems that we always convert `WT_NOTFOUND` to `ENONENT` , given the code at the end of `__wt_session_compact`:

```
done:
    API_END_RET_NOTFOUND_MAP(session, ret);
```

This macro converts any `WT_NOTFOUND` to `ENONENT`:

```
#define API_END_RET_NOTFOUND_MAP(s, ret) \
    API_END(s, ret);                     \
    return ((ret) == WT_NOTFOUND ? ENOENT : (ret))
```

Reply:
Updated in e49caa21d51a4ba29828821d0ef0701d8c0701f0

Reply:
Ah, thanks for checking that!